packages/firestore/exp/src/api/reference.ts

```diff
@@ -233,14 +237,15 @@ export function getDocsFromCache<T>(
 ): Promise<QuerySnapshot<T>> {
   const firestore = cast(query.firestore, FirebaseFirestore);
   const client = ensureFirestoreConfigured(firestore);
+  const userDataWriter = newExpUserDataWriter(firestore, query._converter);
```

Comment:
Optional: you could save a little bit of memory by deferring the creation of the `userDataWriter` until you need it just as you're creating the `QuerySnapshot`.

Reply:
I slightly prefer the current arrangement as I can use the same pattern across all functions. It also keeps the assignment in the Lambda shorter, which makes it fit one in line.